includes/rest-api/class-sensei-rest-api-home-controller.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/**
+ * Sensei Home REST API.
+ *
+ * @package Sensei\Admin
+ * @since   $$next-version$$
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Home REST API endpoints.
+ *
+ * @since $$next-version$$
+ */
+class Sensei_REST_API_Home_Controller extends \WP_REST_Controller {
+
+	/**
+	 * Routes namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Routes prefix.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'home';
+
+
+	/**
+	 * Sensei_REST_API_Home_Controller constructor.
+	 *
+	 * @param string $namespace Routes namespace.
+	 */
+	public function __construct( $namespace ) {
+		$this->namespace = $namespace;
+	}
+
+	/**
+	 * Register the REST API endpoints for Home.
+	 */
+	public function register_routes() {
+		$this->register_get_data_route();
+	}
+
+	/**
+	 * Check user permission for REST API access.
+	 *
+	 * @return bool Whether the user can access the Sensei Home REST API.
+	 */
+	public function can_user_access_rest_api() {
+		return current_user_can( 'manage_sensei' );
+	}
+
+	/**
+	 * Register GET / endpoint.
+	 */
+	public function register_get_data_route() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_data' ],
+					'permission_callback' => [ $this, 'can_user_access_rest_api' ],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get data for Sensei Home frontend.
+	 *
+	 * @return array Setup Wizard data
+	 */
+	public function get_data() {
+
+		return [
+			'tasks'       => [
+				// TODO: Generate based on Setup Wizard data + site info.
+				[
+					'title' => 'Set up Course Site',
+					'done'  => true,
+					'url'   => null,
+				],
+				[
+					'title' => 'Create your first Course',
+					'done'  => false,
+					'url'   => '/wp-admin/edit.php?post_type=course',
+				],
+				[
+					'title' => 'Configure Learning Mode',
+					'done'  => false,
+					'url'   => '/wp-admin/edit.php?post_type=course&page=sensei-settings#course-settings',
+				],
+				[
+					'title' => 'Publish your first Course',
+					'done'  => false,
+					'url'   => '???',
+				],
+			],
+			'quick_links' => [
+				// TODO: Generate or hardcode in frontend.
+				'Courses'           => [
```

Comment:
Not sure about this structure. Is made to fit the usage from frontend, but I was tempted to separate categories from links:
```
'quick_links' => [
  'categories' => ['id' => 'courses', title => 'Courses'],
  'links' => [ ['title' => '...', 'category' => 'course', 'url' => '...'], ....],
  ]
```

Any strong opinions?

Reply:
In the implementation the section titles will need to be i11n-able. Although I doubt it would cause issues, part of me thinks that the title shouldn't come from the key. I'd say either your solution or something like:
```php
'quick_links' => [
	[
		'title' => 'Courses',
		'items' => [
			[
				'title' => 'Import a Course',
				'url'   => 'http://...',
			],
			// ...
		],
	],
	// ...
]
```

Reply:
I also think that having the title as a property instead as being the key is better.

I like Jake's solution here a little more, personally, as it avoids the need to have an extra lockup/filter to get the links from a given category.

Reply:
Changed to what @jom suggested in 71b27dbe3